verification/io/other_defs.gobra

```diff
@@ -122,49 +119,76 @@ type IO_Link adt {
 	IO_PeerOrCore{}
 	IO_NoLink{}
 }
-/* End of Link Types */
-
-/* To clarify, the existence of the following functions seems to be assumed */
-// NOTE:
-// The following functions are provided as parameters to various locales in the Isabelle formalization.
-// However, it is not clear at the moment how these should be translated to Gobra and whether they
-// should have proof obligations attached, or can just be assumed to exist.
 
+// TODO: should be put p1 == dp.GetLocalIA() in the precondition?
+// Should we put dp.Valid() in precondition?
```

Comment:
`dp.Valid()` looks fine to me. I wonder why we return `IO_NoLink{}` when the 1st param is not `dp.GetLocalIA()`. Is it defined like this in Isabelle?  If so, why does it even take two parameters? If looks like the first one is totally unnecessary

Reply:
No this function is not defined in Isabelle. So we have some freedom how we want to define it. At the moment we only store our own linktypes in `DataplaneSpec.` So my reasoning was that we only return a linktype if we call `link_type()`  with our local asid and return `No_link ` as a default value. Alternatively, we could strengthen our `DataplaneSpec` such that it contains all linktypes of the network. This would allow us to write a more general `link_type()` function. Since we never need any other linktypes than our own in the IO-spec I think this might be overkill. What do you think?

Reply:
so, did you check that `link_type` is always passed with the localIA in the place of the 1st param? If so, I would even add, as a precondition, that `p1` is equal to the `dp.GetLocalIA()` and add a comment that, even though we could remove it, we decided to keep it to keep the obvious connection to the Isabelle formalisation.

Maybe even better, we could drop it and add a comment explaining why we dropped the first parameter, this way we do not have to check a precondition on every call to this

Reply:
Dropping it would change quite a lot of other functions. I think we should discuss this in more detail in our next meeting.